library/debug.c

```diff
@@ -80,12 +81,26 @@ void mbedtls_debug_print_msg(const mbedtls_ssl_context *ssl, int level,
     ret = mbedtls_vsnprintf(str, DEBUG_BUF_SIZE, format, argp);
     va_end(argp);
 
-    if (ret >= 0 && ret < DEBUG_BUF_SIZE - 1) {
-        str[ret]     = '\n';
-        str[ret + 1] = '\0';
+    if (DEBUG_BUF_SIZE >= 2) {
```

Comment:
It's not useful to have `DEBUG_BUF_SIZE < 2`:

```
#if defined(static_assert)
static_assert(DEBUG_BUF_SIZE >= 2)
#endif
```

Reply:
(put up around line 72 to avoid making the code harder to read)

Reply:
:p - I have modified it.